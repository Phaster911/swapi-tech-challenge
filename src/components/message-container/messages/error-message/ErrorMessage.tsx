import { RotateCcw } from "react-feather";
import styled from "styled-components";
import StyledMessageSection from "../../../shared/styled-message-wrapper/StyledMessageWrapper";
import { COLORS, controlBorderRadius } from "../../../../style-constants";

const StyledButton = styled.button`
  background-color: ${COLORS.pageBackground};
  border-radius: ${controlBorderRadius};
  border: none;
  padding: 1rem 1.5rem;
  cursor: pointer;
  display: flex;
  gap: 1rem;
`;

type FooterProps = {
  tryAgainHandler: () => void;
};

export default function ErrorMessage({ tryAgainHandler }: FooterProps) {
  return (
    <StyledMessageSection
      aria-labelledby="error-message-title"
      role="alert"
      aria-live="assertive"
    >
      <h2 id="error-message-title">Error: Unable to complete search</h2>
      <p>There was a problem processing your request. Please try again.</p>
      <StyledButton
        id="try-again-button"
        aria-label="try again"
        data-testid="try-again-testid"
        onClick={tryAgainHandler}
      >
        Try Again
        <RotateCcw />
      </StyledButton>
    </StyledMessageSection>
  );
}
