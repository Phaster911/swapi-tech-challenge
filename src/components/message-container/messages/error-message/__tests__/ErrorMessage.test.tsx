import { render, screen, fireEvent } from "@testing-library/react";
import ErrorMessage from "../ErrorMessage";

describe("<ErrorMessage />", () => {
  let tryAgainFn: ReturnType<typeof vi.fn>;

  beforeEach(() => {
    tryAgainFn = vi.fn();
  });

  it("renders the component", () => {
    const { container } = render(<ErrorMessage tryAgainHandler={tryAgainFn} />);
    expect(container).toMatchSnapshot();
  });

  describe("when the try again button is clicked", () => {
    it("calls the tryAgainHandler", () => {
      render(<ErrorMessage tryAgainHandler={tryAgainFn} />);

      expect(screen.getByTestId("try-again-testid")).toBeInTheDocument();
      fireEvent.click(screen.getByTestId("try-again-testid"));
      expect(tryAgainFn).toHaveBeenCalled();
    });
  });
});
