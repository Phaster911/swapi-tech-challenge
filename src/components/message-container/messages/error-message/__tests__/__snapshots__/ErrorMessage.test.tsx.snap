// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<ErrorMessage /> > renders the component 1`] = `
.c0 {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  gap: 1rem;
  text-align: center;
}

.c1 {
  background-color: white;
  border-radius: 5px;
  border: none;
  padding: 0.5rem 0.75rem;
  cursor: pointer;
  display: flex;
  gap: 1rem;
}

<div>
  <div
    aria-labelledby="error-message-title"
    aria-live="assertive"
    class="c0"
    role="alert"
  >
    <h2
      id="error-message-title"
    >
      Error: Unable to complete search
    </h2>
    <p>
      There was a problem processing your request. Please try again.
    </p>
    <button
      aria-label="try again"
      class="c1"
      data-testid="try-again-testid"
      id="try-again-button"
    >
      Try Again
      <svg
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <polyline
          points="1 4 1 10 7 10"
        />
        <path
          d="M3.51 15a9 9 0 1 0 2.13-9.36L1 10"
        />
      </svg>
    </button>
  </div>
</div>
`;
